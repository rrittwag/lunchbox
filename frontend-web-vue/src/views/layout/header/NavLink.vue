<script setup lang="ts">
import { RouterLink } from 'vue-router'

const props = defineProps<{
  to: string
}>()
</script>

<template>
  <li class="inline-block px-2">
    <RouterLink v-slot="{ navigate, isExactActive, route }" :to="props.to" custom>
      <a
        class="nav-link flex h-12 w-12 items-center justify-center"
        :class="isExactActive ? 'text-neutral-800' : 'text-neutral-300'"
        :href="route.path"
        :title="route.meta.title"
        :aria-label="route.meta.title"
        :aria-current="isExactActive ? 'page' : undefined"
        @click="navigate"
      >
        <slot />
      </a>
    </RouterLink>
  </li>
</template>

<style lang="scss">
.nav-link > svg {
  @apply h-8 w-8;
}
</style>
