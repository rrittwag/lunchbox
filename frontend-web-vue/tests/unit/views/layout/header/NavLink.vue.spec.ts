import NavLink from '@/views/layout/header/NavLink.vue'
import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router'
import { render, within } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'

describe('NavLink', () => {
  beforeEach(async () => {
    await router.push('/')
  })

  it('renders', () => {
    const { getByRole } = render(NavLink, {
      props: { to: mockRoute.path },
      slots: { default: 'Link Text' },
      global: {
        plugins: [router],
      },
    })

    const listitem = getByRole('listitem')
    const link = within(listitem).getByRole('link', { name: 'Mock-Route' })
    expect(link).toHaveAttribute('href', mockRoute.path)
    within(listitem).getByText('Link Text')
  })

  it('WHEN route is active THEN has aria-current', async () => {
    const user = userEvent.setup()
    const { getByRole, queryByRole } = render(NavLink, {
      props: { to: mockRoute.path },
      slots: { default: 'Link Text' },
      global: {
        plugins: [router],
      },
    })
    expect(queryByRole('link', { current: 'page' })).not.toBeInTheDocument()

    const link = getByRole('link', { name: 'Mock-Route' })
    await user.click(link)

    expect(queryByRole('link', { current: 'page' })).toHaveAccessibleName('Mock-Route')
  })
})

// --- mocks 'n' stuff

const homeRoute = {
  path: '/',
  meta: { title: 'Home' },
  component: {},
} as unknown as RouteRecordRaw
const mockRoute = {
  path: '/mock-route',
  meta: { title: 'Mock-Route' },
  component: {},
} as unknown as RouteRecordRaw

const router = createRouter({
  history: createWebHistory(),
  routes: [homeRoute, mockRoute],
})
