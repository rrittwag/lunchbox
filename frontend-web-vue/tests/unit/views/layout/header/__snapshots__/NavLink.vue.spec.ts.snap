// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink renders snapshot 1`] = `
<li
  class="inline-block px-2"
>
  
  <a
    aria-label="Mock-Route"
    class="nav-link flex justify-center items-center w-12 h-12 text-neutral-300"
    href="/mock-route"
    title="Mock-Route"
  >
    
    
  </a>
  
</li>
`;
